### ISSUE TITLE
Improve the beat marker behaviour to allow grid stretching

### ISSUE THREAD

**Issue by acolombier** [mixxxdj/mixxx#13768]
Currently, `beats_set_marker` will only add a a new grid, leading the previous grid to have an uneven number of beats. This as to address the case with varying BPM or beat/half-beat track.

Going forward, we add a new CO to allow setting a new marker, which stretches the previous grid and keep complete (that is, contain an integer of beats)

A nice to have would be to make `beats_set_marker` configurable to allow its behaviour to optionally copy `beats_stretch`'s behaviour. This would allow mappings to prioritise the use of `beats_set_marker`, but still allow user to change the default behaviour, in case they don't use tracks with varying BPM or beat phase.

_Originally discussion in https://github.com/mixxxdj/mixxx/issues/13330_

**Comment by daschuer** [comment]
Repsyps has a nice concept:
https://elldev.com/repsyps/adv-grid
It is however slightly different and we have dismissed in an earlier discussion. But IMHO the boundaries still apply. 

It is not an original use case to start a new beat grid at a random position, which is we allow now. This should be possible, but is a new beat with a beat change marker in the left. 

The use case is creating a different beat or bar. This implicit requires a beat change marker at the end of the previous bar. 

So I think the following solution would work:
* beats_set_change_marker shall be always qantize. 
* With slower and faster you can then pick a new tempo 
* We may add a new control for placing beat at the current position. 
* This may change the tempo to the left until the next beatarker and continues the same tempo to the right
* We may also add a beat change marker at the left neighbor in a "single" beat mode.

**Comment by daschuer** [comment]
What do you think?

**Comment by acolombier** [comment]
I disagree with the approach, which is exactly the approach https://github.com/mixxxdj/mixxx/pull/4489 was going for. On paper, it seems to work well and meet the requirement, in practice it's pretty hard to use and is quite not user friendly.

Does anybody know how Serato and Rekordbox deal with that? I know Traktor (<4, before the flexible grid which I haven't tested) is using padding beat.
